internal/db/sqlite/db_update.go

```diff
@@ -587,8 +586,14 @@ func (s *DB) periodicCheckpointLocked(fs []protocol.FileInfo) {
 		if _, err := conn.ExecContext(context.Background(), `PRAGMA journal_size_limit = 67108864`); err != nil {
 			l.Debugln("PRAGMA journal_size_limit:", err)
 		}
-		if _, err := conn.ExecContext(context.Background(), `PRAGMA wal_checkpoint(RESTART)`); err != nil {
+		row := conn.QueryRowContext(context.Background(), `PRAGMA wal_checkpoint(RESTART)`)
+		var res, modified, moved int
+		if row.Err() != nil {
 			l.Debugln("PRAGMA wal_checkpoint(RESTART):", err)
+		} else if err := row.Scan(&res, &modified, &moved); err != nil {
+			l.Debugln("PRAGMA wal_checkpoint(RESTART) (scan):", err)
+		} else {
+			l.Debugln("checkpoint at", s.updatePoints, "returned", res, modified, moved)
 		}
 		s.updatePoints = 0
```

Comment:
Can we do this conditionally based on `res` ?

Reply:
We absolutely could, but I would like to a avoid a regression where we're suddenly running expensive checkpoints all ... the ... time without making progress. Note that auto checkpoint is in effect again as well, as I don't think that had much to do with blocking our own checkpoints here.

Reply:
That's a valid concern. Let's see how things turn out in the beta. We can still be more aggressive if the WAL size keeps being a problem.